package/MDAnalysis/analysis/dssp/dssp.py

```diff
@@ -0,0 +1,404 @@
+    HAS_PYDSSP = False
+    from .pydssp_numpy import (
+        assign,
+        _get_hydrogen_atom_position,
+    )
+
+
+class DSSP(AnalysisBase):
+    """Assign secondary structure using the DSSP algorithm.
+
+    Analyze a selection containing a protein and assign secondary structure
+    using the Kabsch-Sander algorithm :footcite:p:`Kabsch1983`. Only a subset
+    of secondary structure categories are implemented:
+
+    - 'H' represents a generic helix (α-helix, π-helix or :math:`3_{10}` helix)
+    - 'E' represents 'extended strand', participating in beta-ladder (parallel
+      or antiparallel)
+    - '-' represents unordered part
+
+    The implementation was taken from the pydssp package (v. 0.9.0)
+    https://github.com/ShintaroMinami/PyDSSP by Shintaro Minami.
+
+    .. Warning::
+        For the DSSP to work properly, your atoms (either as ``Universe``
+        or ``AtomGroup``) must have only one hydrogen atom "H" per residue,
+        that corresponds to N-bound hydrogen in a peptide bond (or zero hydrogen
+        atoms, if the residue is proline). It is selected using `hydrogen_atom`
+        parameter. It is designed to hande most of the modern syntaxes,
+        but if you encounter an error or unusual results
+        during your run, please report an issue on MDAnalysi
+        `github <https://github.com/MDAnalysis/mdanalysis/issues>`_.
+
+    Parameters
+    ----------
+    atoms : Union[Universe, AtomGroup]
+        input Universe or AtomGroup. In both cases, only protein residues will
+        be chosen prior to the analysis via `select_atoms('protein')`.
+        Heavy atoms of the protein are then selected by name
+        `heavyatom_names`, and hydrogens are selected by name
+        `hydrogen_name`.
+    guess_hydrogens : bool, optional
+        whether you want to guess hydrogens positions, by default ``True``.
+        Guessing is made assuming perfect 120 degrees for all bonds that N
+        atom makes, and a N-H bond length of 1.01 A.
+        If ``guess_hydrogens`` is False, hydrogen atom positions on N atoms
+        will be parsed from the trajectory, except for the "hydrogen" atom
+        positions on PRO residues, and an N-terminal residue.
+    heavyatom_names : tuple[str], default ("N", "CA", "C", "O O1 OT1")
+        selection names that will be used to select "N", "CA", "C" and "O"
+        atom coordinates for the secondary structure determination. The last
+        string contains multiple values for "O" to account for C-term residues.
+    hydrogen_name : str, default "H HN HT1 HT2 HT3"
+        This selection should only select a single hydrogen atom in each residue
+        (except proline), namely the one bound to the backbone nitrogen.
+
+    .. Note::
+       To work with different hydrogen-naming conventions by default, the default
+       selection is broad but if hydrogens are incorrectly selected (e.g., a
+       :exc:`ValueError` is raised) you must customize ``hydrogen_name`` for your
```

Comment:
reference kwargs by single backticks
```suggestion
       :exc:`ValueError` is raised) you must customize `hydrogen_name` for your
```